src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -256,11 +256,34 @@ public override bool HtmlEncode
         }
 
         /// <inheritdoc />
+        public override bool IsBindingAllowed
+        {
+            get
+            {
+                if (MetadataKind == ModelMetadataKind.Property)
```

Comment:
It's defense in depth. I fixed untracked bugs in this exact logic for a similar property in the last PR. It's far easier to just write the correct code in both places.